src/main/pages/user-management/user-management.adoc

```diff
@@ -8,67 +8,102 @@ folder: user-management
 
-Keycloak can create and authenticate users by itself or rely on 3rd party identity management systems and providers.
+Local users and imported federation users must have an email address in their profile.
 
-Note that it is required for all types of users (local or imported from federation) to have the email address field set. Che cannot work with users that have no email in their profile.
+The default Keycloak credentials are `admin:admin`. You can use the `admin:admin` credentials when logging into Che for the first time. It has system privileges. 
 
-Default Keycloak credentials are `admin:admin`. You can find your Keycloak URL either in OpenShift web console > keycloak namespace (if you deployed to OpenShift) of go to `$CHE_HOST:5050/auth` if Che is running on Docker. Admin user is also a pre-defined Che user with privileges on the system scope. So, you may use `admin:admin` credentials when loggin in to Che for the first time.
+To find your Keycloak URL:
 
-[id="che-and-keycloak"]
-== Che and Keycloak
+If Che is deployed on Openshift:
 
-When Che is deployed on OpenShift or installed on Docker, either deployment script or CLI makes sure that Keycloak is properly configured. When a `che-public` client is created, there are two important fields to be populated:
+* Go to OpenShift web console > keycloak namespace 
 
-* *Valid Redirect URIs* need to be the URL you use to access Che. For example, if your original `CHE_HOST` is `1.1.1.1` but you access Che at `myhost` which is an alias, you will see an error from Keycloak, saying that it’s an invalid `redirectURI`. So, if this error occurs, go to Keycloak administration console and check valid Redirect URIs.
-* *Web Origins* - the same concerns Web Origins. An invalid web origin causes CORS error.
+If Che is running on Docker:
+ 
+* Go to `$CHE_HOST:5050/auth`   
 
-[id="keycloak-tokens"]
-== Keycloak Tokens
+[id="configuring-che-to-work-with-keycloak"]
+== Configuring Che to work with Keycloak
 
-By default, a user token gets expired after 30 mins. Keycloak admin can fine tune different realm settings related to Keycloak tokens:
+The deployment script ensures that Keycloak is properly configured when Che is deployed on OpenShift or installed on Docker. When the `che-public` client is created, the following fields are populated:
+
+* *Valid Redirect URIs*: Use this URL to access Che. 
+* *Web Origins*
+
+The following are common errors when configuring Che to work with Keycloak: 
+
+*Invalid `redirectURI` error*:  occurs when you access Che at `myhost`, which is an alias, and your 
+original `CHE_HOST` is `1.1.1.1`.  If this error occurs, go to the Keycloak administration console and the valid redirect URIs are configured.
```

Comment:
It feels like a word is missing here: "and the valid redirect URIs are configured". Perhaps it should be "and ensure that valid redirect URIs are configured"?